pandas/core/dtypes/missing.py

```diff
@@ -644,3 +646,8 @@ def is_valid_na_for_dtype(obj, dtype: DtypeObj) -> bool:
 
     # fallback, default to allowing NaN, None, NA, NaT
     return not isinstance(obj, (np.datetime64, np.timedelta64, Decimal))
+
+
+def _is_string_dtype(dtype: np.dtype) -> bool:
```

Comment:
hmm can you simply move this to dtypes.core.common next to the other one

Reply:
Sure.  most recent commit also added an improvement for `random_text`